collector/collector.go

```diff
@@ -77,13 +77,18 @@ func (c *Collector) Start(ctx context.Context) error {
 	}
 
 	c.appDone = make(chan struct{})
-	go func() {
+	runErr := make(chan error, 1)
+	go func(runErr chan error) {
 		defer close(c.appDone)
-		appErr := c.svc.Run(ctx)
-		if appErr != nil {
-			err = appErr
-		}
-	}()
+		err := c.svc.Run(context.Background())
```

Comment:
Just curious, why is `context.Background()` better than `ctx` here?

Reply:
Good catch that should be the passed context.